### PR TITLE
#111: AJV caching issue

### LINKED ISSUE #111: AJV caching causes validation with stale schema
AJV [caches](https://github.com/ajv-validator/ajv/blob/27178f544fa8b037589f3e04055682b530e8c853/lib/core.ts#L710) the previous compilation of a schema if the reference hasn't changed. This was causing `Property` payloads to be validated with the lookup schema.

### PR DESCRIPTION
**Pull request by mohit-s96**
closes #111

This PR adds changes so that we can keep track of the schemas that have been compiled with a map. We can then re-use the schema or force a recompile based on the map.

### CODE REVIEW

**Comment by darnjo** [comment]
Needs to be brought up to date and tested further before it can be merged. 

Are there tests for the issues that this ticket addresses?

**Comment by mohit-s96** [comment]
>Are there tests for the issues that this ticket addresses?

Yes. Test for this change means we can re-use schema generated once, throughout the process. We were regenerating the schema before each test. I removed that so we only generated schema once and many tests failed (before the changes of this PR). Now after the changes from this PR, we have the tests passing again.

**Comment by darnjo** [comment]
@mohit-s96 - I updated this from the main branch. It looks like the merge is probably what we want but maybe you can take a look before it's approved.

File: lib/schema/utils.js [hunk 1]
```diff
@@ -33,7 +33,14 @@ const processFiles = async ({ inputPath, fileContentsMap }) => {
       const content = await processFile({
         filePath: path.join(inputPath, file)
       });
-      fileContentsMap[fileName] = content;
+      if (fileName.endsWith('.zip')) {
```
Comment (darnjo) [inline lib/schema/utils.js:36]: I think you can do this by checking the first bytes of the file instead?

**Review by darnjo — APPROVED** [review]

**Pull request merged**